kratos/tests/cpp_tests/utilities/test_parallel_utilities.cpp

```diff
@@ -0,0 +1,210 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+
+#include <utility>
+#include <iostream>
+#include "testing/testing.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/openmp_utils.h"
+
+namespace Kratos {
+namespace Testing {
+
+// Basic Type
+KRATOS_TEST_CASE_IN_SUITE(BlockPartitioner, KratosCoreFastSuite)
+{
+    int nsize = 1e3;
+    std::vector<double> data_vector(nsize);
+    for(auto& it : data_vector)
+        it = 5.0;
+
+    //here we raise every entry of a vector to the power 0.1
+    BlockPartition<std::vector<double>>(data_vector).for_each(
+                                         [](double& item)
+    {
+        item = std::pow(item, 0.1);
+    });
+
+    //error check
+    for(auto& item : data_vector)
+    {
+        KRATOS_CHECK_EQUAL(item, std::pow(5.0, 0.1));
+    }
+
+    //shorter form
+    block_for_each(data_vector, [](double& item){
+            item = std::pow(5.0, 0.1);
+    });
+
+    //error check
+    for(auto& item : data_vector)
+    {
+        KRATOS_CHECK_EQUAL(item, std::pow(5.0, 0.1));
+    }
+
+    //here we check for a reduction (computing the sum of all the entries)
+    auto final_sum = BlockPartition<std::vector<double>>(data_vector).for_reduce<SumReduction<double>>(
```

Comment:
Is there a short-hand form for reduction? Something like
```cpp
auto final_sum = block_reduce<SumReduction<double>>(data_vector, [](double &item){ return item; });
```

Reply:
yes, added in test